### PR TITLE
Use defaultValue in currency input

### PR DESCRIPTION
**Pull request by lucasbordeau**
- Fix default value sent to backend, using single quotes by default
- Use default value in field definition and column definition so that field inputs can access it
- Used currency default value in CurrencyFieldInput

### CODE REVIEW

File: packages/twenty-server/src/engine/metadata-modules/field-metadata/utils/validate-default-value-for-type.util.ts [hunk 1]
```diff
@@ -69,13 +69,24 @@ export const validateDefaultValueForType = (
...
     >(validator, conputedDefaultValue as FieldMetadataClassValidation);
 
-    return (
-      validateSync(defaultValueInstance, {
-        whitelist: true,
-        forbidNonWhitelisted: true,
-        forbidUnknownValues: true,
-      }).length === 0
-    );
+    const errors = validateSync(defaultValueInstance, {
```
Comment (lucasbordeau) [inline packages/twenty-server/src/engine/metadata-modules/field-metadata/utils/validate-default-value-for-type.util.ts:72]: @magrinj Is it ok or shall we use a logger ?

File: packages/twenty-front/src/modules/object-metadata/utils/getDefaultValueForBackend.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+export const getDefaultValueForBackend = (
+  defaultValue: any,
+  fieldMetadataType: FieldMetadataType,
+) => {
+  if (fieldMetadataType === FieldMetadataType.Currency) {
+    const currencyDefaultValue = defaultValue as FieldCurrencyValue;
+    return {
+      amountMicros: currencyDefaultValue.amountMicros,
+      currencyCode: `'${defaultValue.currencyCode}'` as any,
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-metadata/utils/getDefaultValueForBackend.ts:12]: use currencyDefaultValue here too to be consistent?

File: packages/twenty-front/src/modules/object-metadata/utils/getDefaultValueForBackend.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { FieldCurrencyValue } from '@/object-record/record-field/types/FieldMetadata';
+import { FieldMetadataType } from '~/generated-metadata/graphql';
+
+export const getDefaultValueForBackend = (
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-metadata/utils/getDefaultValueForBackend.ts:4]: Reading the code, I'm not sure about it, i think this should go to the BE
I wonder if the backend should actually do the work and expect a different kind of defaultValue per FieldType.
If the fieldType is CurrencyCode, we don't care about uuid(), now() functions indeed, we just want currencyCode (string) + amountMicros (int)

File: packages/twenty-server/src/engine/metadata-modules/field-metadata/utils/validate-default-value-for-type.util.ts [hunk 4]
```diff
@@ -69,13 +69,24 @@ export const validateDefaultValueForType = (
...
+      whitelist: true,
+      forbidNonWhitelisted: true,
+      forbidUnknownValues: true,
+    });
+
+    const hasErrors = errors.length > 0;
+
+    if (hasErrors) {
+      console.error(
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/metadata-modules/field-metadata/utils/validate-default-value-for-type.util.ts:81]: yes use the logger. I would return { isValid, errors } here and log where this util is used using the logger. It should not be the responsability of this util to log. It should either return the errors or throw them

File: packages/twenty-front/src/modules/object-metadata/utils/getDefaultValueForBackend.ts (hunk 3 above)
Comment (charlesBochet) [reply packages/twenty-front/src/modules/object-metadata/utils/getDefaultValueForBackend.ts:4]: I'm in favor of moving forward with the current implementation and take the time to refactor next week, should be quite light

**Review by charlesBochet — APPROVED** [review]

File: packages/twenty-server/src/engine/metadata-modules/field-metadata/utils/validate-default-value-for-type.util.ts (hunk 1 above)
Comment (magrinj) [reply packages/twenty-server/src/engine/metadata-modules/field-metadata/utils/validate-default-value-for-type.util.ts:87]: I think we're ok here, but what you've done is only logging in the console.
If you want to be more specific on the error sent by the backend to the front-end, you can host the errors in the `IsFieldMetadataDefaultValue`, I'll give you an example below.

File: packages/twenty-server/src/engine/metadata-modules/field-metadata/validators/is-field-metadata-default-value.validator.ts [hunk 5]
```diff
@@ -14,13 +14,17 @@ import {
   FieldMetadataType,
 } from 'src/engine/metadata-modules/field-metadata/field-metadata.entity';
 import { validateDefaultValueForType } from 'src/engine/metadata-modules/field-metadata/utils/validate-default-value-for-type.util';
+import { LoggerService } from 'src/engine/integrations/logger/logger.service';
 
 @Injectable()
 @ValidatorConstraint({ name: 'isFieldMetadataDefaultValue', async: true })
 export class IsFieldMetadataDefaultValue
```
Comment (magrinj) [inline packages/twenty-server/src/engine/metadata-modules/field-metadata/validators/is-field-metadata-default-value.validator.ts:21]: In this class you can do something like that, if you want the error sent to the front-end to be more specific:

```typescript
@Injectable()
@ValidatorConstraint({ name: 'isFieldMetadataDefaultValue', async: true })
export class IsFieldMetadataDefaultValue
  implements ValidatorConstraintInterface
{
  private validationErrors: string[] = [];

  constructor(private readonly fieldMetadataService: FieldMetadataService) {}

  async validate(
    value: FieldMetadataDefaultValue,
    args: ValidationArguments,
  ): Promise<boolean> {
    // Try to extract type value from the object
    let type: FieldMetadataType | null = args.object['type'];

    if (!type) {
      // Extract id value from the instance, should happen only when updating
      const id: string | undefined = args.instance?.['id'];

      if (!id) {
        return false;
      }

      let fieldMetadata: FieldMetadataEntity;

      try {
        fieldMetadata = await this.fieldMetadataService.findOneOrFail(id);
      } catch {
        return false;
      }

      type = fieldMetadata.type;
    }

    const { isValid, errors } = validateDefaultValueForType(type, value);

    if (errors.length > 0) {
      this.validationErrors = errors.map((error) => error.toString());
    }

    return isValid;
  }

  defaultMessage(): string {
    if (this.validationErrors.length > 0) {
      return this.validationErrors.join(', ');
    }

    return 'FieldMetadataDefaultValue is not valid';
  }
}
```

**Pull request merged**